config/pipeline.yaml

```diff
@@ -0,0 +1,200 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+# Configuration for running the complete pipeline including modeling and evaluation.
+
+# Absolute paths to non-code resources.
+file_paths:
+  queries: "/home/jupyter/github_msba_capstone/msba-capstone-2020/queries"
+
+# Global config used by multiple pipeline steps. Also made available
+# as parameters to all SQL queries.
+global:
+  # Project and dataset configuration.
+  source_project_id: "bigquery-public-data"
+  destination_project_id: "ut-goog"
+  source_dataset: "cfpb_complaints"
+  destination_dataset: "cfpb_complaints"
+
+  # Clean data, minor refined versions of provided data.
+  source_table: "complaint_database"
+  nlp_features_table: "complaints_nlp"
+  cleaned_features_table: "complaints_clean_categories"
+  nulls_removed_table: "complaints_nulls_removed"
+  clean_table: "complaints_clean"
+  
+  # Split features into training and prediction.
+  train_predict_split: "TrainPredictSplit"
+  features_train_table: "FeaturesTrain"
+  features_predict_table: "FeaturesPredict"
+
+  # Output tables for predictions and failed predictions from the batch
+  # prediction job, ex. numeric column recieved a string.
+  predictions_table: "Predictions"
+  failed_predictions_table: "FailedPredictions"
+  
+  # Evaluation
+  eval_table: "Evaluation"
+  eval_metrics_table: "EvaluationMetrics"
+
+  # AutoML Parameters. It is assumed that no other datasets or models have
+  # been created with these names, or the pipeline will fail. They can be
+  # deleted using the client, or the UI.
+  dataset_display_name: "CFPBDataset"
+  model_display_name: "CFPBModel"
+  automl_compute_region: "us-central1"
+
+  # Final model will be trained on test data, metrics are currently based
+  # on the predict split instead. AutoML Requires at least one row of test data
+
+# Files with templated queries, in file_paths.queries.
+query_files:
+  remove_nulls: "remove_nulls.sql"
+  clean_categories: "clean_categories.sql"
+  combine_tables: "combine_tables.sql"
+  train_predict_split: "train_and_predict_split.sql"
+  prediction_features: "prediction_features.sql"
+  training_features: "training_features.sql"
+
+# Parameters specific to individual SQL pipeline steps.
+query_params:
+
+  removing_nulls:
+    column1: "consumer_complaint_narrative"
+    column2: "company_response_to_consumer"
+    
+  category_cleanup:
```

Comment:
It would be better to restructure this. First, it's not essential that this cleanup be parameterized -- it's okay to write this assuming the table is the table as it is (this could apply to the null cleaning as well, though if the SQL is so generic then its a good idea to have an arbitrary null-cleaning functionality). With that said, if you want to keep it parameterized rather than hard coding cleanup the whole old_x new_x thing is not really useful as config, and you should look into a more flexible configuration structure (nested lists? I'm not quite sure).

Reply:
That makes sense. I have hard-coded the category cleanup and removed it from the config file. However, I have kept the null-cleaning as a generic functionality.